packages/mui-material-next/src/ButtonGroup/ButtonGroup.test.tsx

```diff
@@ -181,24 +184,25 @@ describe('<ButtonGroup />', () => {
   });
 
   it('should forward the context to children', () => {
-    let context;
+    let context: ButtonGroupContextType;
     render(
       <ButtonGroup size="large" variant="contained">
         <ButtonGroupContext.Consumer>
           {(value) => {
             context = value;
+            return <Button />;
           }}
         </ButtonGroupContext.Consumer>
       </ButtonGroup>,
     );
-    expect(context.variant).to.equal('contained');
-    expect(context.size).to.equal('large');
-    expect(context.fullWidth).to.equal(false);
-    expect(context.disableRipple).to.equal(false);
-    expect(context.disableTouchRipple).to.equal(false);
-    expect(context.disableElevation).to.equal(false);
-    expect(context.disabled).to.equal(false);
-    expect(context.color).to.equal('primary');
+    expect(context!.variant).to.equal('contained');
```

Comment:
```suggestion
    expect(context?.variant).to.equal('contained');
```

Reply:
I have a naive question - how does `context!.variant` even work though ?! 😅 

![Screenshot 2023-11-08 at 5 05 40 PM](https://github.com/mui/material-ui/assets/4997971/e813127c-8745-4dc4-a5d7-2daa9bf512cd)

Reply:
The problem is that when I use `context?.variant`, ts complains that the variable `context` was used before initialization. TS does not understand that the `render`-function is executed before the `expect`-functions.

Reply:
Oh it's the TS "non null assertion operator" https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator